core/trino-main/src/main/java/io/trino/operator/HashAggregationOperator.java

```diff
@@ -70,6 +74,8 @@
         private final SpillerFactory spillerFactory;
         private final JoinCompiler joinCompiler;
         private final BlockTypeOperators blockTypeOperators;
+        private final PartialAggregationControlFactory partialAggregationControlFactory;
+        private final PartialAggregationControl partialAggregationControl;
```

Comment:
this should be `Optional`. Not every aggregation is partial

Reply:
> Optional or null here makes code more convoluted as it needs to be handled on every access.

If you use `Optional` then you don't have to have `PartialAggregationControl` or `PartialAggregationControlFactory` as interfaces (since there is just single implementation). There isn't really going to be another implementation and it won't be pluggable really. Hence, an interface just to have noop is an overkill.